py/src/nupic/examples/tm/hello_tm.py

```diff
@@ -0,0 +1,135 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2013, Numenta, Inc.
+#               2019, David McDougall
+#
+# Unless you have an agreement with Numenta, Inc., for a separate license for
+# this software code, the following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+__doc__ = """
+This program shows how to access the Temporal Memory algorithm directly.  This
+program demonstrates how to create a TM instance, train it, get predictions and
+anomalies, and inspect the state.
+
+The code here runs a very simple version of sequence learning, with one
+cell per column. The TM is trained with the simple sequence A->B->C->D->E
+"""
+
+from nupic.bindings.sdr import SDR
+from nupic.algorithms import TemporalMemory as TM
+
+# Utility routine for printing an SDR in a particular way.
+def formatBits(sdr):
+  s = ''
+  for c in range(sdr.size):
+    if c > 0 and c % 10 == 0:
+      s += ' '
+    s += str(sdr.dense.flatten()[c])
+  s += ' '
+  return s
+
+def printStateTM( tm ):
+    # Useful for tracing internal states
+    print("Active cells     " + formatBits(tm.getActiveCells()))
+    print("Winner cells     " + formatBits(tm.getWinnerCells()))
+    tm.activateDendrites(True)
+    print("Predictive cells " + formatBits(tm.getPredictiveCells()))
+    print("Anomaly", tm.anomaly * 100, "%")
+    print("")
+
+
+print("################################################################################")
+print(__doc__)
+print("################################################################################")
+print("")
+print("Creating the Temporal Memory")
+tm = TM(columnDimensions = (50,),
+        cellsPerColumn=1,
```

Comment:
TM with 1 cell/col does not really make sense (no higher order learning), unless it's necessary for the example, this should use "normal" number of cells, ie 4,8,...